spec/std/socket/spec_helper.cr

```diff
@@ -12,7 +23,17 @@ def each_ip_family(&block : Socket::Family, String, String ->)
     block.call Socket::Family::INET, "127.0.0.1", "0.0.0.0"
   end
 
-  describe "using IPv6" do
-    block.call Socket::Family::INET6, "::1", "::"
+  if SocketSpecHelper.supports_ipv6?
+    describe "using IPv6" do
+      block.call Socket::Family::INET6, "::1", "::"
+    end
+  else
+    pending "using IPv6"
+  end
+end
+
+def unused_local_port
+  TCPServer.open("::", 0) do |server|
```

Comment:
I suppose this should use an IPv4 address, too?

Reply:
Using `::` instead of `0.0.0.0` seems fine to be fine for some reason. At least in the environment I could  test this in. What fails  is binding or connecting to `::1` instead of `127.0.0.1`. So I left this alone in order to minimize  changes.

Reply:
So, if this works, why are all the `TCPServer.new` and `TCPServer.open` in the specs changed with explicit `127.0.0.1` address?

Reply:
Because I wanted to preserve intent and they were binding to `::1` (equivalent to `127.0.0.1`), not `::` (equivalent to `0.0.0.0`).

Reply:
@waj it works, but only for `::` -> `0.0.0.0`, and not `::1` -> `127.0.0.1`.

Reply:
I mean, you replaced `TCPServer.new(0)` with `TCPServer.new("127.0.0.1", 0)`. The default binds to `::` https://github.com/crystal-lang/crystal/blob/bab17be529a40a6b3175dfd07b629cc0752df7b5/src/socket/tcp_server.cr#L58

Also, seems like if `TCPServer.new(0)` doesn't work on non IPv6 environments it's a bug we should fix

Reply:
https://github.com/jhass/crystal/runs/744998680?check_suite_focus=true#step:7:23152

🤷 Maybe I'm wrong that `::` always works?

Reply:
mm... apparently `TCPServer.new(0)` works just fine, but then `local_address` returns an IPv6 address that `TCPClient` cannot use

Reply:
makes sense, it's fine for binding but not for connecting.